tools/webhook-apicoverage/webhook/apicoverage_recorder.go

```diff
@@ -0,0 +1,137 @@
+	Logger *zap.SugaredLogger
+	ResourceForest resourcetree.ResourceForest
+	ResourceMap map[schema.GroupVersionKind]webhook.GenericCRD
+	NodeRules resourcetree.NodeRules
+	FieldRules resourcetree.FieldRules
+	DisplayRules view.DisplayRules
+}
+
+// Init initializes the resources trees for set resources.
+func (a *APICoverageRecorder) Init() {
+	for resourceKind, resourceObj := range a.ResourceMap {
+		a.ResourceForest.AddResourceTree(resourceKind.Kind, reflect.ValueOf(resourceObj).Elem().Type())
+	}
+}
+
+// RecordResourceCoverage updates the resource tree with the request.
+func (a *APICoverageRecorder) RecordResourceCoverage(w http.ResponseWriter, r *http.Request) {
+	var body []byte
+	if data, err := ioutil.ReadAll(r.Body); err != nil {
+		a.Logger.Errorf("Failed reading request body: %v", err)
+		a.writeAdmissionResponse(a.getAdmissionResponse(false, "Admission Denied"), w)
+		return
+	} else {
+		body = data
+	}
+
+	review := &v1beta1.AdmissionReview{}
+	if _, _, err := decoder.Decode(body, nil, review); err != nil {
+		a.Logger.Errorf("Unable to 	decode request: %v", err)
+		a.writeAdmissionResponse(a.getAdmissionResponse(false, "Admission Denied"), w)
+		return
+	}
+
+	gvk := schema.GroupVersionKind {
+		Group:   review.Request.Kind.Group,
+		Version: review.Request.Kind.Version,
+		Kind:    review.Request.Kind.Kind,
+	}
+	if err := json.Unmarshal(review.Request.Object.Raw, a.ResourceMap[gvk]); err != nil {
+		a.Logger.Error("Failed unmarshalling review.Request.Object.Raw for type: %s Error: %v", a.ResourceMap[gvk], err)
+		a.writeAdmissionResponse(a.getAdmissionResponse(false, "Admission Denied"), w)
+		return
+	}
+	resourceTree := a.ResourceForest.TopLevelTrees[gvk.Kind]
+	resourceTree.UpdateCoverage(reflect.ValueOf(a.ResourceMap[gvk]).Elem())
+	a.writeAdmissionResponse(a.getAdmissionResponse(true, "Welcome Aboard"), w)
+}
+
+func (a *APICoverageRecorder) getAdmissionResponse(allowed bool, message string) (*v1beta1.AdmissionResponse) {
+	return &v1beta1.AdmissionResponse{
+		Allowed: allowed,
+		Result: &v1.Status{
+			Message: message,
+		},
+	}
+}
+
+func (a *APICoverageRecorder) writeAdmissionResponse(admissionResp *v1beta1.AdmissionResponse, w http.ResponseWriter) {
+	responseInBytes, err := json.Marshal(admissionResp)
+	if err != nil {
+		a.Logger.Errorf("Failing mashalling review response: %v", err)
+	}
+
+	if _, err := w.Write(responseInBytes); err != nil {
+		a.Logger.Errorf("%v", err)
+	}
+}
+
+// GetResourceCoverage retrieves resource coverage data for the passed in resource via query param.
+func (a *APICoverageRecorder) GetResourceCoverage(w http.ResponseWriter, r *http.Request) {
+	resource := r.URL.Query().Get(ResourceQueryParam)
+	if _, ok := a.ResourceForest.TopLevelTrees[resource]; !ok {
+		fmt.Fprintf(w, "Resource information not found for resource: %s", resource)
+		return
+	}
+
+	var ignoredFields coveragecalculator.IgnoredFields
+	ignoreFieldsFilePath := os.Getenv("KO_DATA_PATH") + "/ignoredfields.yaml"
```

Comment:
suggest renaming to ignoredFieldsFilePath

Reply:
Done.